process/analysis.py

```diff
@@ -99,27 +158,48 @@ def analysis(r):
         os.makedirs(r.config['region_dir'])
 
     archive_parameters(r, settings)
+    
+    """Conditional Google Earth Engine authentication """
+    try:
+        if r.config['gee'] is True:
+            authenticate_gcloud_and_gee(r)
+        else:
+         print("\nGoogle Earth Engine authentication skipped as 'gee' is set to False in configuration file.")
+    except KeyError:
+        print("\nGoogle Earth Engine authentication skipped as 'gee' key is missing in the configuration.")
 
     print_autobreak(
         f"\nAnalysis time zone: {settings['project']['analysis_timezone']} (to set time zone for where you are, edit config.yml)\n\n",
     )
     start_analysis = time.time()
     print(f"Analysis start:\t{time.strftime('%Y-%m-%d_%H%M')}")
-    study_region_setup = {
-        '_00_create_database.py': 'Create database',
-        '_01_create_study_region.py': 'Create study region',
-        '_02_create_osm_resources.py': 'Create OpenStreetMap resources',
-        '_03_create_network_resources.py': 'Create pedestrian network',
-        '_04_create_population_grid.py': 'Align population distribution',
-        '_05_compile_destinations.py': 'Compile destinations',
-        '_06_open_space_areas_setup.py': 'Identify public open space',
-        '_07_locate_origins_destinations.py': 'Analyse local neighbourhoods',
-        '_08_destination_summary.py': 'Summarise spatial distribution',
-        '_09_urban_covariates.py': 'Collate urban covariates',
-        '_10_gtfs_analysis.py': 'Analyse GTFS Feeds',
-        '_11_neighbourhood_analysis.py': 'Analyse neighbourhoods',
-        '_12_aggregation.py': 'Aggregate region summary analyses',
-    }
+    # Dynamically construct study_region_setup based on r.config['gee']
+    study_region_setup = [
+        ('_00_create_database.py', 'Create database'),
+        ('_01_create_study_region.py', 'Create study region'),
+        ('_02_create_osm_resources.py', 'Create OpenStreetMap resources'),
+        ('_03_create_network_resources.py', 'Create pedestrian network'),
+        ('_04_create_population_grid.py', 'Align population distribution'),
+        ('_05_compile_destinations.py', 'Compile destinations'),
+        ('_06_open_space_areas_setup.py', 'Identify public open space'),
+    ]
+    # Conditionally include step 7 & 8
+    if r.config.get('gee', False):

```

Comment:
This should work, but a simpler and perhaps more flexible approach would be to use the previous formatting but add in a new step like `xx_earth_engine_indicators.py: 'Calculate optional indicators using Google Earth Engine'`.  Then, the earth engine configuration could be checked in there and whatever indicators there might be (there might be more in future) might be triggered in some workflow.  Or else, its not configured and skipped. 

That would save hard coding indicators here --- there are concepts but not specific indicators, ideally.  That way, when new indicators are added, we can avoid major change to the overall workflow.  Ideally, new indicators would be added through configuration, not code, as far as possible.  So this could be an approach that moves towards that ideal.